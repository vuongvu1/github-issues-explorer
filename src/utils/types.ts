export type IssueType = Array<{
  id: string;
  title: string;
  number: number;
  createdAt: string;
  author: {
    login: string;
  };
  comments: {
    totalCount: number;
  };
}>;
