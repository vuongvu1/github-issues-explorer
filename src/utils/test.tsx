import { ReactNode } from "react";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import { ThemeProvider } from "styled-components";
import { render } from "@testing-library/react";
import { MockedProvider } from "@apollo/react-testing";
import { BrowserRouter } from "react-router-dom";
import { theme } from "src/globalStyles";
import {
  IssueState,
  IssueOrderField,
  OrderDirection,
} from "src/generated/graphql";

const mockStore = configureMockStore();
const store = mockStore({
  repoSlice: {
    url: "https://github.com/reactjs/reactjs.org",
    currentPage: 1,
    searchParams: {
      name: "reactjs.org",
      owner: "reactjs",
      status: IssueState.Open,
      orderBy: IssueOrderField.CreatedAt,
      orderDir: OrderDirection.Desc,
      before: undefined,
      after: undefined,
      first: 10,
      last: undefined,
    },
  },
});

export const customRender = (Component: ReactNode) =>
  render(
    <ThemeProvider theme={theme}>
      <MockedProvider mocks={[]} addTypename={false}>
        <BrowserRouter>
          <Provider store={store}>{Component}</Provider>
        </BrowserRouter>
      </MockedProvider>
    </ThemeProvider>
  );
