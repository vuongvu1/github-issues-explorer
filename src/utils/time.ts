const ONE_YEAR_IN_SECONDS = 31536000;
const ONE_MONTH_IN_SECONDS = 2592000;
const ONE_DAY_IN_SECONDS = 86400;
const ONE_HOUR_IN_SECONDS = 3600;
const ONE_MINUTE_IN_SECONDS = 60;

export const timeSince = (date: Date) => {
  const seconds = Math.floor((new Date().getTime() - date.getTime()) / 1000);

  let interval = seconds / ONE_YEAR_IN_SECONDS;

  if (interval > 1) {
    return `${Math.floor(interval)} years`;
  }

  interval = seconds / ONE_MONTH_IN_SECONDS;
  if (interval > 1) {
    return `${Math.floor(interval)} months`;
  }

  interval = seconds / ONE_DAY_IN_SECONDS;
  if (interval > 1) {
    return `${Math.floor(interval)} days`;
  }

  interval = seconds / ONE_HOUR_IN_SECONDS;
  if (interval > 1) {
    return `${Math.floor(interval)} hours`;
  }

  interval = seconds / ONE_MINUTE_IN_SECONDS;
  if (interval > 1) {
    return `${Math.floor(interval)} minutes`;
  }

  return `${Math.floor(seconds)} seconds`;
};
