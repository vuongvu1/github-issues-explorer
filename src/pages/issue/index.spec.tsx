import { cleanup } from "@testing-library/react";
import { customRender } from "src/utils/testUtils";
import IssuePage from "./index";

describe("IssuePage", () => {
  afterEach(cleanup);

  test("should match snapshot", () => {
    const { container } = customRender(<IssuePage />);
    expect(container).toMatchSnapshot();
  });
});
