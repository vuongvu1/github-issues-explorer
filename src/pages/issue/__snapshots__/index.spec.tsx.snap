// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IssuePage should match snapshot 1`] = `
<div>
  <div
    class="sc-crHlIS jkdymW"
  >
    <div
      class="sc-egiSv kwgRtk"
    >
      <h2
        class="sc-jRQAMF iYUFVx"
        type="h2"
      >
        reactjs / reactjs.org - Issues
      </h2>
    </div>
    <div
      class="sc-bqiQRQ ihGzzV"
    >
      <div
        class="sc-ksdxAp kHMzp"
      >
        <div
          class="sc-furvIG joAWHN"
        >
          <div
            class="sc-pVTma spJgp"
          >
            <div
              class="sc-dJjZJu fotGxU"
            >
              <svg
                class="sc-gsDJrp hiSWTT"
                height="48"
                viewBox="0 0 48 48"
                width="48"
              >
                <circle
                  cx="24"
                  cy="4"
                  fill="#000"
                  r="4"
                />
                <circle
                  cx="12.19"
                  cy="7.86"
                  fill="#000"
                  r="3.7"
                />
                <circle
                  cx="5.02"
                  cy="17.68"
                  fill="#000"
                  r="3.4"
                />
                <circle
                  cx="5.02"
                  cy="30.32"
                  fill="#000"
                  r="3.1"
                />
                <circle
                  cx="12.19"
                  cy="40.14"
                  fill="#000"
                  r="2.8"
                />
                <circle
                  cx="24"
                  cy="44"
                  fill="#000"
                  r="2.5"
                />
                <circle
                  cx="35.81"
                  cy="40.14"
                  fill="#000"
                  r="2.2"
                />
                <circle
                  cx="42.98"
                  cy="30.32"
                  fill="#000"
                  r="1.9"
                />
                <circle
                  cx="42.98"
                  cy="17.68"
                  fill="#000"
                  r="1.6"
                />
                <circle
                  cx="35.81"
                  cy="7.86"
                  fill="#000"
                  r="1.3"
                />
              </svg>
            </div>
            <div
              class="sc-jrQzUz gRUHIY"
            >
              <div
                class="sc-iqsfdx FyauU"
              >
                <svg
                  aria-hidden="true"
                  class="prefix__octicon prefix__octicon-issue-opened"
                  height="1em"
                  viewBox="0 0 16 16"
                  width="1em"
                >
                  <title>
                    Open Issues
                  </title>
                  <path
                    d="M8 1.5a6.5 6.5 0 100 13 6.5 6.5 0 000-13zM0 8a8 8 0 1116 0A8 8 0 010 8zm9 3a1 1 0 11-2 0 1 1 0 012 0zm-.25-6.25a.75.75 0 00-1.5 0v3.5a.75.75 0 001.5 0v-3.5z"
                    fill-rule="evenodd"
                  />
                </svg>
                 Open
              </div>
              <div
                class="sc-iqsfdx duBOUF"
              >
                <svg
                  aria-hidden="true"
                  class="prefix__octicon prefix__octicon-check"
                  height="1em"
                  viewBox="0 0 16 16"
                  width="1em"
                >
                  <title>
                    Closed Issues
                  </title>
                  <path
                    d="M13.78 4.22a.75.75 0 010 1.06l-7.25 7.25a.75.75 0 01-1.06 0L2.22 9.28a.75.75 0 011.06-1.06L6 10.94l6.72-6.72a.75.75 0 011.06 0z"
                    fill-rule="evenodd"
                  />
                </svg>
                 Closed
              </div>
            </div>
            <div
              class="sc-kDThTU EEUgj"
            >
              No issues found.
            </div>
          </div>
          <div
            class="sc-gKckTs jLoZpw"
          >
            <div>
              <button
                class="sc-iCfLBT eDmwTq"
                data-testid="prev-btn"
              >
                Previous
              </button>
              <button
                class="sc-iCfLBT kbrdGG"
              >
                1
              </button>
              <button
                class="sc-iCfLBT eDmwTq"
                data-testid="next-btn"
              >
                Next
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
